buildSrc/src/integTest/groovy/org/elasticsearch/gradle/YamlRestCompatTestPluginFuncTest.groovy

```diff
@@ -0,0 +1,114 @@
+        }
+        """
+
+        buildFile << """
+        plugins {
+          id 'elasticsearch.yaml-rest-compat-test'
+        }
+        """
+
+        when:
+        def result = gradleRunner("yamlRestCompatTest").build()
+
+        then:
+        result.task(':yamlRestCompatTest').outcome == TaskOutcome.NO_SOURCE
+        result.task(':copyRestApiCompatSpecsTask').outcome == TaskOutcome.NO_SOURCE
+        result.task(':copyRestApiCompatTestTask').outcome == TaskOutcome.NO_SOURCE
+    }
+
+    def "yamlRestCompatTest executes and copies api and tests from :bwc:minor"() {
+        given:
+        String api = "wrongversion.json"
+        String test = "wrongversion.yml"
+        setupInternalRestResources(api, test)
+
+        addSubProject(":distribution:bwc:minor") <<  """
+        configurations { checkout }
+        artifacts {
+            checkout(new File(projectDir, "checkoutDir"))
+        }
+        """
+
+        //add the compatible test and api files
+        file("distribution/bwc/minor/checkoutDir/src/yamlRestTest/resources/rest-api-spec/test/foo/10_basic.yml") << ""
+        file("distribution/bwc/minor/checkoutDir/rest-api-spec/src/main/resources/rest-api-spec/api/foo.json") << ""
+
+        buildFile << """
+          apply plugin: 'elasticsearch.yaml-rest-compat-test'
+          import org.elasticsearch.gradle.testclusters.TestDistribution;
+          testClusters {
+            yamlRestCompatTest.setTestDistribution(TestDistribution.INTEG_TEST)
+          }
+        """
+
+        when:
+        def result = gradleRunner("yamlRestCompatTest").build()
+
+        then:
+        result.task(':yamlRestCompatTest').outcome == TaskOutcome.NO_SOURCE //no Java classes in source set
+        result.task(':copyRestApiCompatSpecsTask').outcome == TaskOutcome.SUCCESS
+        result.task(':copyRestApiCompatTestTask').outcome == TaskOutcome.SUCCESS
+
+        File resourceDir = new File(testProjectDir.root, "build/resources/yamlRestCompatTest/rest-api-spec")
+        // ensure the compatible resources from bwc:minor are in the resource dir
+        assert new File(resourceDir, "test/foo/10_basic.yml").exists()
+        assert new File(resourceDir, "api/foo.json").exists()
+        // we don't copy the "normal" rest resources to the compat resource dir
+        assert new File(resourceDir, "/test/" + test).exists() == false
+        assert new File(resourceDir, "/api/" + api).exists() == false
+
+        when:
+        //add the mock java test and dependency to run, the runner from the yamlRestTest sourceSet is used
+        file("src/yamlRestTest/java/MockIT.java") << "import org.junit.Test;class MockIT { @Test public void doNothing() { }}"
+
+        buildFile << """
+           dependencies {
+           yamlRestTestImplementation "junit:junit:4.12"
+        }
+        """
+
+        result = gradleRunner("yamlRestCompatTest", '-i').buildAndFail() //expect failure - we don't actually spin up a cluster
+
+        then:
```

Comment:
What exactly are we testing in this scenario? Just that the copy tasks are still up to date? If not, should this be a separate feature method?

Reply:
The first `when` is testing that copy is correct, and the second `when` that test attempts to spin up a cluster and execute. 

Testing that the copy is up to date is probably more about the behavior of the copy resource plugin, not this plugin. (We don't have tests for that yet, but we probably should, just not in this PR).

If you think is a confusing, i can move the config from the second `when` up to the first, and not test the up to date behaviour here.

Reply:
Yeah, I'm wondering if this makes for a good integration test given that it's actually failing to do so. We're just testing that it "attempts" to do so and that might be quicker and easier to validate in a unit test. That is use something like `ProjectBuilder` to validate that applying the plugin does what we expect, like creates the source set, the corresponding test cluster, etc.

@breskeby thoughts?

Reply:
yeah that might be better tested in a unit test if we think its required